src/components/app-login/test/app-login.spec.tsx

```diff
@@ -51,5 +56,35 @@ describe('app-login', () => {
     //   expect(alertMock).toHaveBeenCalledTimes(1);
     //   // expect(global.alert).toBeCalledWith()
     //});
+
+    it('shows warning message if email is invalid', async () => {
+      const page2 = await newSpecPage({
+        components: [AppLogin],
+        html: `<app-login></app-login>`,
+      });
+      const data = {
+        user: 'saif123gmailcom',
+        password: '13asb',
+      };
+      page2.rootInstance.onSubmit(data);
```

Comment:
use the constant data declared above everywhere instead of creating new one

Reply:
we cant declared above because they are different